external/storm-hdfs/src/main/java/org/apache/storm/hdfs/trident/HdfsState.java

```diff
@@ -69,63 +78,92 @@
+            } else if (this.rotationPolicy instanceof FileSizeRotationPolicy) {
+                long rotationBytes = ((FileSizeRotationPolicy) rotationPolicy).getMaxBytes();
+                LOG.warn("FileSizeRotationPolicy specified with {} bytes.", rotationBytes);
+                LOG.warn("Recovery will fail if data files cannot be copied within topology.message.timeout.secs.");
+                LOG.warn("Ensure that the data files does not grow too big with the FileSizeRotationPolicy.");
+            } else if (this.rotationPolicy instanceof TimedRotationPolicy) {
+                LOG.warn("TimedRotationPolicy specified with interval {} ms.", ((TimedRotationPolicy) rotationPolicy).getInterval());
+                LOG.warn("Recovery will fail if data files cannot be copied within topology.message.timeout.secs.");
+                LOG.warn("Ensure that the data files does not grow too big with the TimedRotationPolicy.");
+            }
             if (this.fsUrl == null) {
                 throw new IllegalStateException("File system URL must be specified.");
             }
             this.fileNameFormat.prepare(conf, partitionIndex, numPartitions);
             this.hdfsConfig = new Configuration();
-            Map<String, Object> map = (Map<String, Object>)conf.get(this.configKey);
-            if(map != null){
-                for(String key : map.keySet()){
+            Map<String, Object> map = (Map<String, Object>) conf.get(this.configKey);
+            if (map != null) {
+                for (String key : map.keySet()) {
                     this.hdfsConfig.set(key, String.valueOf(map.get(key)));
                 }
             }
-            try{
+            try {
                 HdfsSecurityUtil.login(conf, hdfsConfig);
                 doPrepare(conf, partitionIndex, numPartitions);
                 this.currentFile = createOutputFile();
 
-            } catch (Exception e){
+            } catch (Exception e) {
                 throw new RuntimeException("Error preparing HdfsState: " + e.getMessage(), e);
             }
 
-            if(this.rotationPolicy instanceof TimedRotationPolicy){
-                long interval = ((TimedRotationPolicy)this.rotationPolicy).getInterval();
-                this.rotationTimer = new Timer(true);
-                TimerTask task = new TimerTask() {
-                    @Override
-                    public void run() {
-                        try {
-                            rotateOutputFile();
-                        } catch(IOException e){
-                            LOG.warn("IOException during scheduled file rotation.", e);
-                        }
-                    }
-                };
-                this.rotationTimer.scheduleAtFixedRate(task, interval, interval);
+            if (this.rotationPolicy instanceof TimedRotationPolicy) {
+                ((TimedRotationPolicy) this.rotationPolicy).start();
+            }
```

Comment:
Actually the warnings are specific to HDFS state and by moving it to the rotation policy we make the rotation policy aware of the recovery which might not be ideal if the rotation policy class needs to be reused in some other context.

Moving the start logic to policy specific prepare methods makes sense. I will make this change.